### DISCUSSION TITLE
Streaming mode needed

### DISCUSSION THREAD

**Discussion by Xeladoes** [ppy/osu#27074] [Ideas] [answered]
The desired feature to be added: 
Feature 1: Hide national flag.
Feature 2: Hide nickname( or change it to something else because we need chat).
Feature 3: Hide chat(From a livestreaming perspective, adding a toggle switch around the chat to indicate whether the chat is displayed would be very beneficial. However, this would greatly impact the normal user experience, which is something we do not want to see. So the design of this feature or whether to add it may require further consideration. No modifications may be necessary since the current comment reporting system is highly effective.).

For certain reasons, broadcasting with nicknames containing certain words or displaying certain national flags may not be allowed under certain circumstances and may result in banning. I have noticed that when trying to broadcast osu!stable or osu!lazer, the broadcast room of osu!lazer is often banned while osu!stable is not. Therefore, having these two features would greatly improve this situation.

Of course, all of the above are just my personal suggestions; I believe and have also seen that many others share the same needs. Whether to add this feature depends on whether osu!lazer wants to broadcast in regions affected by complex political factors.

(FYI. In China, broadcasting osu!lazer will definitely result in being banned. I have communicated with customer service several times regarding the reasons for the banning of my livestreams. The only solution provided is to cover the content in OBS (which is very complicated and significantly affects the live streaming quality.) or to modify image files. However, I am not sure how to modify them, as osu!lazer files are stored using hashes, and I am unaware of which file I need to modify.
I really enjoy the online mode of osu!lazer and I would love to share and I am sharing my joyful gaming experiences with everyone. Most of the Chinese osu players I know mainly play osu!stable, which, in my personal experience, is indeed smoother and more stable. However, the restrictions from the broadcasting platform prevent me from doing so (sharing lazer), and this makes me feel frustated.)

**Comment by bdach** [comment, marked as answer]
> Feature 1: Hide national flag.

https://github.com/ppy/osu/discussions/20154
https://github.com/ppy/osu/discussions/24369 (https://github.com/ppy/osu/discussions/24369#discussioncomment-6587699 especially relevant I guess)

**Reply by Nana-Sakura** [reply]
Based on local legislations in China, Streaming must:
- Standardize the Five-Star red flag. (Showing Flag of Taiwan, the Province of China is also unacceptable in most cases.)
- Filter any content without review, especially chat. (Sometimes there may be inappropriate content in some players' names, Streaming will also be interrupted at once.)

Comparing applying changes to the contents with simply hiding them, latter may be better.

**Reply by Xeladoes** [reply]
For today I met two players who came into the room and directly made insulting remarks about Chinese martyrs, and the report took effect very quickly, which is great, but it's best not to show up this kind of content, because the efficiency of China's live broadcast bans is very high with the support of cv algorithms. That's why we need some feature like this.

**Comment by OwenCMYK** [comment]
This would also benefit:
- VTubers
- Regular streamers
- Players who don't want their irl friends knowing their online account
- Players who may be embarrassed by their username and don't want to show it to others

I am strongly in support of this feature for those reasons, as well as the China-related reasons.

**Comment by Kotoki1337** [comment]
almost every few months there's a discussion or issues like this. very early on, ppy denying this feature multiple times:

[#12547 (reply)](https://github.com/ppy/osu/issues/12547#issuecomment-825629759)
[#22089 (reply)](https://github.com/ppy/osu/discussions/22089#discussioncomment-4630897)
also the discussion that bdach mentioned above

i think ppy's point is no need to made a feature for a certain country. and also they are the core elements of osu!. but it's obviously that requirements for this feature is high.
